src/exprutils.jl

```diff
@@ -3,7 +3,10 @@
 using Core: MethodInstance
 using Base: MethodList
 
-const poppable_macro = (Symbol("@inline"), Symbol("@noinline"), Symbol("@propagate_inbounds"), Symbol("@eval"))
+const poppable_macro = (Symbol("@inline"), Symbol("@noinline"), Symbol("@propagate_inbounds"), Symbol("@eval"), Symbol("@pure"))
+
+is_poppable_macro(ex) = ex ∈ poppable_macro ||
+    ex.head == :. && ex.args[1] == :Base && ex.args[2].value ∈ poppable_macro
```

Comment:
What if `ex` isn't an expression? Add `if ex isa Expr &&` to the front of this line. Also add parenthesis to disambiguate the grouping of logical operators.